framework/widgets/ActiveField.php

```diff
@@ -930,7 +930,10 @@ protected function addAriaAttributes(&$options)
      */
     protected function addErrorClassIfNeeded(&$options)
     {
-        if ($this->model->hasErrors($this->attribute)) {
+        // Get proper attribute name when attribute name is tabular.
+        $attributeName = yii\helpers\BaseHtml::getAttributeName($this->attribute);
```

Comment:
We do not rely directly on `Base%` helpers. Use `Html::getAttributeName()`, as it is done above in this file `https://github.com/yiisoft/yii2/pull/15805/files#diff-afe28192283fbfadaf6145620191c339R904`